### PR TITLE
Fix exception when pushing down TRY expression 

### PR DESCRIPTION
**Pull request by pranjalssh**
Let's fix 2 bugs in master. Coming from https://github.com/prestodb/presto/pull/19313

Commit 1: Previous commit fixed a bug in `isTryFunction`. Previously, it would always result to false, because it was comparing QualifiedObjectName vs String. After fixing, we saw some crashes in PredicatePushdown. As previously, all queries were running with it false, I just removed the code which checked that. Added a test for it that we don't pushdown expressions with TRY()

Commit 2: Fixes planner timeouts on large queries. Don't copy an expensive map. ExpressionAnalyzer is called for every expression, and outerScopeSymbolTypes has types for whole plan. So just don't copy. 

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java [hunk 1]
```diff
@@ -268,7 +268,7 @@ private ExpressionAnalyzer(
         this.parameters = requireNonNull(parameters, "parameterMap is null");
         this.isDescribe = isDescribe;
         this.warningCollector = requireNonNull(warningCollector, "warningCollector is null");
-        this.outerScopeSymbolTypes = ImmutableMap.copyOf(requireNonNull(outerScopeSymbolTypes, "outerScopeSymbolTypes is null"));
+        this.outerScopeSymbolTypes = requireNonNull(outerScopeSymbolTypes, "outerScopeSymbolTypes is null");
```
Comment (NikhilCollooru) [inline presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java:271]: Nice ! How did you figure out that this copy is causing the timeout ?

**Comment by NikhilCollooru** [comment]
Can you run some verifier tests and confirm that the changes will resolve the failures we saw ?

**Comment by NikhilCollooru** [comment]
Can you copy the description for each commit as well. Like what you described above..will be great if added to the commit message.

**Comment by pranjalssh** [comment]
@NikhilCollooru  I ran the offending queries locally with PrismQueryRunner. Verifier tests are random anyway, given this was caught 1 week after push... So I'm fairly confident with this fix.

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java (hunk 1 above)
Comment (pranjalssh) [reply presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java:271]: Added test. Just run locally with same query and pause debugger at random points to see what its doing

**Review by NikhilCollooru — APPROVED** [review]
LGTM

**Pull request merged**